docs/configuration.md

```diff
@@ -0,0 +1,457 @@
+>##
+>>### config.playback.options
+>>##### Type: `PKPlaybackOptionsObject`
+>>```js
+>>{
+>>  html5: {
+>>    hls: Object,
+>>    dash: Object
+>>  }
+>>}
+>>```
+>>##### Default: 
+>>```js
+>>{
+>>  html5: {
+>>    hls: {},
+>>    dash: {}
+>>  }
+>>}
+>>```
+>>##### Description: Defined the media source adapters configurations.
+>>For `hls` configuration, see [hls.js](https://github.com/video-dev/hls.js/blob/master/doc/API.md#fine-tuning) documentation.
+>For `dash` configuration, see [shaka-player](https://shaka-player-demo.appspot.com/docs/api/tutorial-config.html) documentation.
+>## 
+>>### config.playback.preferNative
+>>##### Type: `PKPreferNativeConfig`
+>>```js
+>>{
+>>  hls: boolean,
+>>  dash: boolean
+>>}
+>>```
+>>##### Default: 
+>>```js
+>>{
+>>  hls: false,
+>>  dash: false
+>>}
+>>```
+>>##### Description: Whether to prefer native browser playback (if supported) with media source extensions.
+>>If one of the values sets to true and the player chooses to play the truthly media source extension, the player will try to play it natively if supported by the browser. 
+>> #### Example:
+>>Lets assume the following configuration:
+>>```js
+>>var config = {
+>>  playback: {
+>>    preferNative:  {
+>>      hls: true
+>>    }
+>>  }
+>>};
+>>```
+>> In case we're running in _Safari_ browser, the player will play native hls playback managed by the _Safari_ browser. But in case we're running in browser which hls playback is not supported natively, _Chrome_ for instance, the player will play hls using the `hls.js` library.
+>## 
+>>### config.playback.streamPriority
+>>##### Type: `Array<PKStreamPriorityObject`
+>>>##### Type `PKStreamPriorityObject`
+>>>```js
+>>>{
+>>>   engine: string,
+>>>   format: string
+>>>}
+>>>```
+>>##### Default: 
+>>```js
+>>[
+>>   {
+>>     engine: "html5",
+>>     format: "hls"
+>>   },
+>>   {
+>>     engine: "html5",
+>>     format: "dash"
+>>   },
+>>   {
+>>     engine: "html5",
+>>     format: "progressive"
+>>    }
+>>]
+>>```
+>>##### Description: Specifies the {engine, stream format} tuple priority of the player by ascending order.
+>>As soon as the player receives his sources, he will go over this array, and will try to play a source with a matched stream format from a matched engine. 
+>>For example, for the above priority configuration, the player will try to play hls stream using an html5 engine first. If an hls stream isn't received, he will continue to play dash stream using an html5 engine. If a dash stream isn't received, he will continue to play progressive stream using an html5 engine.
+
+### Workflow between `sources`, `preferNative` & `streamPriority`
+After we have gained knowledge of the different options available in the player configuration, let's see the results for what format will be played for the following configuration of the player:
```

Comment:
Now that we've learned about the different options available in the player configuration, let's see the results for which format will be played for the following configuration of the player: